fv3core/testing/validation.py

```diff
@@ -0,0 +1,87 @@
+import inspect
+from typing import Callable, Tuple
+
+import numpy as np
+
+import fv3core.stencils.updatedzd
+
+
+def get_selective_class(
+    cls,
+    selective_arg_names,
+    origin_domain_func: Callable[..., Tuple[Tuple[int, ...], Tuple[int, ...]]],
+):
+    """
+    Convert a model class into one that sets nans on non-validated outputs,
+    and gives a helper function to retrieve the output subset we want to validate.
+
+    Using this ensures that if these non-validated values are ever used, a test
+    will fail because an output will have NaN.
+    """
+
+    class SelectivelyValidated:
+        """
+        Wrapper class that sets non-validated outputs to nan, and gives a helper
+        function to retrieve the output subset to be validated.
+        """
+
+        def __init__(self, *args, **kwargs):
+            self.wrapped = cls(*args, **kwargs)
+            origin, domain = origin_domain_func(self.wrapped)
```

Comment:
could alternatively say `self.origin, self.domain=` and then not have assign them below

Reply:
I removed the assignments below instead, because the attributes aren't used anywhere.